src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -414,17 +808,11 @@ def add_service(
 
         self.services[str(service.id)] = service
 
-        if validate:
-            # If validate is set, trigger the initial compile immediately, and fill in all
-            # the default values.
-            self.compile(model=None, service_id=service.id, validation=True, add_defaults=True)
-
     def compile(
         self,
         model: typing.Optional[str] = None,
         service_id: typing.Optional[uuid.UUID] = None,
         validation: bool = True,
-        add_defaults: bool = False,
     ) -> None:
```

Comment:
These two are noticeable changes which break the (unreleased) behavior introduced yesterday.  I did it for the following reasons:
1. I believe that adding defaults should not be done during (or right before) a compile, as this is not how the api handles it, but rather when the service is created or updated.  We didn't have support for create/update operations yesterday, but we have now.  So I moved this logic in the create/update methods.
2. Given that we now have explicit create/update operations, I think it makes no sense to have the possibility to trigger a validation compile when adding the service, this is actually more the use case of the create method.  So I reverted back that change.